src/utils/dplasma_arena_datatype.h

```diff
@@ -11,7 +11,7 @@
 #include "parsec/arena.h"
 #include "parsec/class/parsec_hash_table.h"
 
-#define REUSE_ARENA_DATATYPE
+#undef REUSE_ARENA_DATATYPE
```

Comment:
Did you check performance that changes brings?

Reply:
The reuse code is completely broken, so it was never really used. This patch fixes all concurrent accesses issues, but the reuse has issues even without concurency. There are two major issues with the reuse:
1. hash key contains the pointer to the data, so an arena could never be reused between taskpools.
2. ignoring the above, all arenas are deleted at the end of a taskpool, so they never persist long enough to give the opportunity to another taskpool to reuse them.